website/static/js/apiApplication.js

```diff
@@ -381,6 +387,37 @@ var ApplicationDetailViewModel = oop.extend(ChangeMessageMixin, {
             }
         });
     },
+    resetSecret: function () {
+        var appData = this.appData();
+        var self = this;
+        bootbox.confirm({
+            title: 'Reset client secret?',
+            message: language.apiOauth2Application.resetSecretConfirm,
+            callback: function (confirmed) {
+                if (confirmed){
+                    var request = self.client.resetOne(appData);
+                    request.done(function (dataObj) {
+                        self.appData().clientSecret(dataObj.clientSecret());
+                        self.changeMessage(
+                            language.apiOauth2Application.dataUpdated,
+                            'text-success',
+                            5000);
+                    }.bind(self));
+                    request.fail(function (xhr, status, error) {
+                        $osf.growl('Error',
+                            language.apiOauth2Application.resetSecretError,
+                            'danger');
+
+                        Raven.captureMessage('Error resetting instance secret', {
+                            url: self.apiResetUrl,
+                            status: status,
+                            error: error
```

Comment:
Per the style guide, we want to customize the confirmation dialog:
1. This is a dangerous action, so it should not be blue
2. The text should be more specific to the action (it should not say "ok"; "Reset secret" might be appropriate)

https://centerforopenscience.github.io/osf-style/style.html#buttons
https://centerforopenscience.github.io/osf-style/ui.html#modals